### PR TITLE
First version CRD DOCS

### PR DESCRIPTION
**Pull request by idanl21**
- First version of CRD DOCS
- change busybox image to opster-pub-ecr image (from rate limit errors on past image)

### CODE REVIEW

File: opensearch-operator/pkg/builders/cluster.go [hunk 1]
```diff
@@ -260,7 +260,7 @@ func NewSTSForNodePool(cr *opsterv1.OpenSearchCluster, node opsterv1.NodePool, v
 
 func DockerImageForCluster(cr *opsterv1.OpenSearchCluster) string {
 	// TODO: Determine version based on CR
-	return "opensearchproject/opensearch:1.2.3"
+	return "opensearchproject/opensearch:1.0.0"
```
Comment (dbason) [inline opensearch-operator/pkg/builders/cluster.go:263]: I think this is dead code that could be removed.

File: docs/designs/crd.md [hunk 2]
```diff
@@ -1 +1,159 @@
...
+A [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) is an extension of the Kubernetes API, many core Kubernetes functions are now built using custom resources, making Kubernetes more modular.
+Cluster admins can update custom resources independently of the cluster itself. Once a custom resource is installed, users can create and access its objects using kubectl, just as they do for built-in resources like Pods.
+
+The CustomResourceDefinition API resource allows you to define custom resources. Defining a CRD object creates a new custom resource with a name and schema that you specify. The Kubernetes API serves and handles the storage of your custom resource. Every resource is build from `KGV` that stands for Group Version Resource and this is what drives the Kubernetes API Server structure. 
+The `OpensearchCLuster` CRD is representing an Opensearch cluster.
+
+
+Our CRD is Defined by kind: `OpenSearchCluster`,group: `opensearch.opster.io` and version `v1`.
+<table>
```
Comment (dbason) [inline docs/designs/crd.md:14]: Any reason to use a html table over a markdown one?

File: docs/designs/crd.md [hunk 3]
```diff
@@ -1 +1,293 @@
...
+        <td>false</td>
+      </tr><tr>
+        <td><b>security</b></td>
+        <td>object</td>
+        <td>Defined security reconciler configuration</td>
+        <td>false</td>
+      </tr><tr>
+        <td><b>nodePools</b></td>
+        <td>[]object</td>
+        <td>The nodePools object is build from nodePool objects and define different nodePools  in OpensearchCluster. Thats the main resource that responsible for deploying OpensearchCluster Statefulsets  </td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:92]:
```suggestion
        <td>List of objects that define the different nodePools in an OpensearchCluster. Each nodePool represents a group of nodes with the same opensearch roles and resources. Each nodePool is deployed as a Kubernetes StatefulSet. Together they form the opensearch cluster.</td>
```

File: docs/designs/crd.md [hunk 4]
```diff
@@ -1 +1,293 @@
...
+        <td>true</td>
+      </tr><tr>
+        <td><b>Dashboards</b></td>
+        <td>object</td>
+        <td>Opensearch-dashboards configuration</td>
+        <td>false</td>
+      </tr><tr>
+        <td><b>confMgmt</b></td>
+        <td>object</td>
+        <td>Responsible about enabling additional OpensaerchOperator components (reconcilers)  </td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:82]:
```suggestion
        <td>Config object to enable additional OpensearchOperator features/components</td>
```

File: docs/designs/crd.md [hunk 5]
```diff
@@ -1 +1,293 @@
...
+    <tbody><tr>
+        <td><b>httpPort</b></td>
+        <td>int32</td>
+        <td>http exposure port</td>
+        <td>false</td>
+        <td>9200</td>
+      </tr><tr>
+        <td><b>vendor</b></td>
+        <td>string</td>
+        <td>Vendor declaration (sts docker image is built for it)</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:123]:
```suggestion
        <td>Vendor distribution to use for the cluster, currently only opensearch is supported</td>
```

File: docs/designs/crd.md [hunk 6]
```diff
@@ -1 +1,293 @@
...
+        <td>false</td>
+        <td>opensearch</td>
+      </tr><tr>
+        <td><b>version</b></td>
+        <td>string</td>
+        <td> Opensearch applicative version  </td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:129]:
```suggestion
        <td>Version of opensearch to deploy</td>
```

File: docs/designs/crd.md [hunk 7]
```diff
@@ -1 +1,293 @@
...
+      </tr><tr>
+        <td><b>ServiceAccount</b></td>
+        <td>string</td>
+        <td>k8s service account name</td>
+        <td>false</td>
+        <td>cluster name</td>
+      </tr><tr>
+        <td><b>ServiceName</b></td>
+        <td>string</td>
+        <td>k8s service name</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:141]:
```suggestion
        <td>Name to use for the k8s service to expose the cluster internally</td>
```

File: docs/designs/crd.md [hunk 8]
```diff
@@ -1 +1,293 @@
...
+        <td><b>SetVMMaxMapCount</b></td>
+        <td>bool</td>
+        <td>will add VMmaxMapCount</td>
+        <td>false</td>
+        <td></td>
+      </tr><tr>
+        <td><b>ExtraConfig</b></td>
+        <td>string</td>
+        <td>Added extra items to opensearch.yml</td>
+        <td>string</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:154]:
```suggestion
        <td>map[string]string</td>
```

File: docs/designs/crd.md [hunk 9]
```diff
@@ -1 +1,293 @@
...
+        <td>Opensearch-dashboards version</td>
+        <td>false</td>
+        <td>latest</td>
+      </tr><tr>
+      </tr><tr>
+        <td><b>Tls</b></td>
+        <td>DashboardsTlsConfig</td>
+        <td>defining Dashbaord TLS configuration</td>
+        <td>false</td>
+        <td>false</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:205]:
```suggestion
        <td></td>
```

File: docs/designs/crd.md [hunk 10]
```diff
@@ -1 +1,293 @@
...
+        <td><b>version</b></td>
+        <td>string</td>
+        <td>Opensearch-dashboards version</td>
+        <td>false</td>
+        <td>latest</td>
+      </tr><tr>
+      </tr><tr>
+        <td><b>Tls</b></td>
+        <td>DashboardsTlsConfig</td>
+        <td>defining Dashbaord TLS configuration</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:203]:
```suggestion
        <td>TLS-related configuration for Dashboards</td>
```

File: docs/designs/crd.md [hunk 11]
```diff
@@ -1 +1,293 @@
...
+        <td>false</td>
+      </tr><tr>
+</table>
+
+
+<h3 id="GeneralConfig">
+  NodePools
+</h3>
+
+Every NodePool is defining different Opensearch Nodes StatefulSet
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:214]:
```suggestion
Each NodePool represents a group of nodes in the OpenSearch cluster that have the same role and resources. They are deployed as Kubernetes StatefulSets
```

File: docs/designs/crd.md [hunk 12]
```diff
@@ -1 +1,293 @@
...
+    <tbody><tr>
+        <td><b>component</b></td>
+        <td>string</td>
+        <td>statefulset name - will create $cluster-name-$component STS </td>
+        <td>true</td>
+        <td>-</td>
+      </tr><tr>
+        <td><b>replicas</b></td>
+        <td>int</td>
+        <td>defines NodePool deployment's replicas</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:235]:
```suggestion
        <td>Number of nodes/replicas in the pool</td>
```

File: docs/designs/crd.md [hunk 13]
```diff
@@ -1 +1,293 @@
...
+        <td>true</td>
+        <td>1</td>
+      </tr><tr>
+        <td><b>diskSize</b></td>
+        <td>string</td>
+        <td> nodePool data disk size </td>
+        <td>true</td>
+        <td> - </td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:243]:
```suggestion
        <td>30Gi</td>
```

File: docs/designs/crd.md [hunk 14]
```diff
@@ -1 +1,293 @@
...
+        <td><b>replicas</b></td>
+        <td>int</td>
+        <td>defines NodePool deployment's replicas</td>
+        <td>true</td>
+        <td>1</td>
+      </tr><tr>
+        <td><b>diskSize</b></td>
+        <td>string</td>
+        <td> nodePool data disk size </td>
+        <td>true</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:242]:
```suggestion
        <td>false</td>
```
Is not required

File: docs/designs/crd.md [hunk 15]
```diff
@@ -1 +1,293 @@
...
+        <td> - </td>
+      </tr><tr>
+        <td><b>NodeSelector</b></td>
+        <td>map[string]string</td>
+        <td>add NodeSelector to nodePool</td>
+        <td>false</td>
+        <td> - </td>
+      </tr><tr>
+      </tr><tr>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:251]: NodePool has no TLS config, the entire table row is wrong here

File: docs/designs/crd.md [hunk 16]
```diff
@@ -1 +1,293 @@
...
+        <td>defining Dashbaord TLS configuration</td>
+        <td>false</td>
+        <td>false</td>
+      </tr><tr>
+      </tr><tr>
+        <td><b>resources</b></td>
+        <td>corev1.ResourceRequirements</td>
+        <td> Define NodePool resources </td>
+        <td>false</td>
+        <td>Default Opensearch resources</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:263]:
```suggestion
        <td></td>
```
If not set uses empty resources (meaning no requests/limits on the pods)

File: docs/designs/crd.md [hunk 17]
```diff
@@ -1 +1,293 @@
...
+      </tr><tr>
+      </tr><tr>
+        <td><b>roles</b></td>
+        <td>[]string </td>
+        <td> define Opensearch roles to the nodeGroup </td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:268]:
```suggestion
        <td>List of OpenSearch roles to assign to the nodePool</td>
```

File: docs/designs/crd.md [hunk 18]
```diff
@@ -1 +1,293 @@
...
+        <td>true</td>
+        <td> - </td>
+      </tr><tr>
+      </tr><tr>
+        <td><b>JVM</b></td>
+        <td>string</td>
+        <td> declare JVM</td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:275]:
```suggestion
        <td>JVM args. Use this to define heap size (recommendation: Set to half of memory request)</td>
```

File: docs/designs/crd.md [hunk 19]
```diff
@@ -1 +1,293 @@
...
+        <td>false</td>
+        <td> half of the STS requests  </td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:277]:
```suggestion
        <td>-Xmx512M -Xms512M</td>
```

File: docs/designs/crd.md [hunk 20]
```diff
@@ -1 +1,293 @@
...
+        <td>false</td>
+        <td>cluster name</td>
+      </tr><tr>
+        <td><b>SetVMMaxMapCount</b></td>
+        <td>bool</td>
+        <td>will add VMmaxMapCount</td>
+        <td>false</td>
+        <td></td>
+      </tr><tr>
+        <td><b>ExtraConfig</b></td>
```
Comment (swoehrl-mw) [inline docs/designs/crd.md:151]:
```suggestion
        <td><b>additionalConfig</b></td>
```

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
This is missing some fields:
* nodePool.persistence
* nodePool.additionalConfig
* confMgmt
* general.drainDataNodes
* general.defaultRepo
* Fields from ImageSpec in general
* DashboardsTlsConfig
* Security and its subobjects are completely missing
* probably some more I didn't see

Types `object` should be the respective type, e.g. `[]NodePool` instead of `[]object`.

All field names should be written exactly as they appear in the YAML (e.g. `jvm` instead of `JVM`, `tolerations` instead of `Tolerations`).

**Pull request merged**